### ISSUE TITLE
Can't get it working with precompiled assets

### ISSUE THREAD

**Issue by jayrbolton** [browserify-rails/browserify-rails#52]
Everything works beautifully in my development env without asset precompilation, but as soon as I do `rake assets:precompile`, all pages give me `require is undefined` and none of the Browserify wrapper code is present inside my `public/assets` js files

I've tried every application.rb config option listed in the readme, and I've tried tweaking my `config.assets.precompile` to have *every* file in /app get precompiled. No dice. I've also followd all the steps here: http://ryochikazawa.com/2015/01/30/heroku-with-browserify-rails.html

I'm using rails 3.2.18, sprockets 2.2.3

Not sure how & where to troubleshoot this. Have messed with `browserify_processor.rb` a bunch, but my understanding of Tilt is limited

**Comment by cymen** [comment]
Are you certain your browserify configuration is present in the production environment file? Asset compilation runs under that environment. One one project I'm using browserify-rails on, it looks like this:

```sh

**Comment by cymen** [comment]
Although thinking about it -- it probably isn't that. Could it be that in your Gemfile, you are requiring `browserify-rails` under `:development` but not `:production`?

**Comment by jayrbolton** [comment]
Thanks for the quick reply! I've got browserify 0.7.1 definitely installed correctly, including in production. It does make sens that it's not set up in production. I tried `RAILS_ENV=development rake assets:precompile`, still no luck.

Also tried adding `config.browserify_rails.force = true` directly in config/environments/production.rb and no luck, along with a line below to force precompilation of every asset in the app.

Some other things I tried before:
- added some log statements in `prepare` and `evaluate` inside browserify_processor.rb -- they print correctly in development server logs without precompilation. They don't print at all on rake assets:precompile in any environment
- Inside production.rb, `config.browserify_rails` looks to be instantiated correctly and looks okay:
```
- :force: false
- :paths:
 - !ruby/object:Proc {}
  - !ruby/object:Proc {}
   - :node_bin: node_modules/.bin/
   - :evaluate_node_modules: false
   - :source_map_environments:
    - development
    - :use_browserifyinc: true
```

**Comment by cymen** [comment]
Hrm... I need an instance with the same Rails and Sprockets version at this point to try to figure it out. I can't do that right now but I can look later. Thinking about it though in case I missed something. Do you have anything in `config/browserify.yml`?

**Comment by jayrbolton** [comment]
We've got nothing in `config/browserify.yml`. I'll try updating sprockets and going to the latest version of Rails 3.2 to see if that helps

**Comment by jayrbolton** [comment]
Narrowed it down a bit: for some reason rake is not running the `:setup_browserify` initializer inside the railtie in the gem. It runs everything before and after, but the initializer isn't run. I believe that's my problem. It *does* get run by my rails server.

**Comment by jayrbolton** [comment]
Okay, I finally figured it out. It was the one line in my `config/application.rb` that I somehow failed to notice, even though I messed with pretty much everything else in that file (just one of those days :sob:)
```rb
config.assets.initialize_on_precompile = false
```

That was causing no initializers to run. I added that because of some Devise documentation from a long past bygone era of our app that I'm pretty sure is no longer relevant.

Thanks again for the help!
